agent_build_refactored/tools/runner.py

```diff
@@ -0,0 +1,910 @@
+# Copyright 2014-2022 Scalyr Inc.
```

Comment:
This is a refactored version of the  [agent_build/tools/environment_deployments/deployments.py](https://github.com/scalyr/scalyr-agent-2/blob/master/agent_build/tools/environment_deployments/deployments.py)

I fully refactored everything here, that's why I thought it would be easier to see it as a new file.
It basically implements the same main classes, `Deployment` now  `Runner`, `DeploymentStep` is `RunnerStep`